### DISCUSSION TITLE
outlook.live.com: ads

### DISCUSSION THREAD

**Discussion by dasbootbee** [uBlockOrigin/uAssets#17566] [Q&A]
### Prerequisites

- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Use this [button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out.)
- [X] I forced an update of my filter lists. (Click the ["Purge all caches"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches) button while holding the 'Shift' key, then click the ["Update now"](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now) button.)
- [X] I did not remove any of the [default filter lists](https://user-images.githubusercontent.com/585534/91296476-ab9d8c00-e76a-11ea-9fb9-87d6861bdf73.png), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable external filter lists, or I have verified that the issue still occurs without enabling external filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkpoints.

### URL address of the web page

`https://outlook.live.com/mail/0/inbox/id/AQQkADAwATNiZmYAZC04M2FlLWFiOGYtMDACLTAwCgAQAJleFyrS0fdJjszFS71Q4xU%3D`

### Category

ads

### Description

Same issue as #15953

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![screenshot](https://user-images.githubusercontent.com/1261402/231600549-8160f58a-92f1-410e-a99c-2341860a8417.png)

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.48.4
Chromium: 112
filterset (summary): 
  network: 185395
  cosmetic: 218323
  scriptlet: 39215
  html: 0
listset (total-discarded, last-updated): 
  added: 
    adguard-annoyance: 67163-95, now
    adguard-generic: 69795-520, now
    adguard-social: 20241-849, now
    adguard-spyware-url: 975-2, now
    curben-phishing: 66243-2, never
    curben-pup: 192-0, never
    fanboy-annoyance: 80515-6507, now
    fanboy-cookiemonster: 46571-46571, now
    fanboy-social: 21052-21052, now
    fanboy-thirdparty_social: 69-20, now
    ublock-annoyances: 4935-254, now
  default: 
    user-filters: 3-0, never
    easylist: 64394-873, now
    easyprivacy: 30812-1371, now
    plowe-0: 3750-950, now
    ublock-abuse: 78-0, now
    ublock-badware: 6589-364, now
    ublock-filters: 33713-4122, now
    ublock-privacy: 294-6, now
    ublock-quick-fixes: 371-222, never
    ublock-unbreak: 1976-91, now
    urlhaus-1: 7420-3, now
filterset (user): [array of 3 redacted]
trustedset: 
  added: [array of 10 redacted]
switchRuleset: 
  added: [array of 1 redacted]
modifiedUserSettings: 
  cloudStorageEnabled: true
modifiedHiddenSettings: [none]
supportStats: 
  allReadyAfter: 437 ms (selfie)
  maxAssetCacheWait: 216 ms
popupPanel: 
  blocked: 174
  blockedDetails: 
    live.com: 5
    bing.com: 2
    github.com: 5
    microsoft.com: 54
    msn.com: 1
    office.net: 107
```
</details>

**Comment by Procyon-b** [comment]
Verify that you haven't disabled cosmetic filtering for the site.

Anyway, this should work:
```
outlook.live.com##.customScrollBar.jEpCF > div > div.full:not([id])
```

**Comment by dasbootbee** [comment]
Cosmetic filtering is not disabled.  

I just added the filter to My Filters and refreshed the page and I still see the ad emails, unfortunately.

**Comment by krystian3w** [comment]
On PC/iMac/MacBook you can test my suggestions:

https://github.com/uBlockOrigin/uAssets/issues/15953#issuecomment-1501894560


> ```adb
> ! ad at top in 2-nd column:
> outlook.live.com##div[role="complementary"] > div:has(> * .fbAdCallToAction)
> !
> ! :
> ```
> ![obraz](https://user-images.githubusercontent.com/35370833/230922003-18a34d4b-e2ed-400d-9cc4-95b840f6c28a.png)
> 
> ---
> 
> ```adb
> ! ad at bottom about MS 365 upgrade:
> outlook.live.com###MainModule .disableTextSelection.customScrollBar + div button
> !
> ! mobile but rows are hardcoded by style attr in absolute positioning:
> outlook.live.com###app [role="main"] div[role="group"]:has([role="link"])
> outlook.live.com###app [role="main"] article:has([role="link"])
> outlook.live.com###app [role="main"] [style*="relative"] [style*="absolute"] + div[style*="absolute"]:style(position: static !important;)
> ! so needed is redesign mobile layout to static.
> ```

As well as you don't already have any advertising preferences for the outlook/hotmail account (https://github.com/uBlockOrigin/uAssets/issues/15953#issuecomment-1502636735) then we rather need a demo account to see how much MS has changed the site design of this part of the users.

**Reply by dasbootbee** [reply]
Correct -- there is no advertising preference setting.

**Reply by Yuki2718** [reply]
No need for demo page, I have accounts and it's easy to create one, but I can't reproduce. If anything, View config can affect.

**Reply by krystian3w** [reply]
But as your accounts have "efficient" options for managing the ad system and freshly created ones too, they are useless - we need to bring up one without the option to "control" ads.

**Reply by Yuki2718** [reply]
What are "efficient" options? I have several accouts some of them are very old.

**Reply by krystian3w** [reply]
`https://outlook.live.com/mail/0/options/general/advertisement` - "recommended"/"faster" layout doesen't have ad above e-mails.

https://github.com/AdguardTeam/AdguardFilters/issues/137637#issuecomment-1357631179

**Reply by Yuki2718** [reply]
Same as OP, I don't see such an option.

**Comment by krystian3w** [comment]
And check these modified <i>@</i>**Procyon-b** suggestion in Picker:

```adb
##.customScrollBar > div > div.full:not([id])
```

the `.jEpCF` can be unstable.

**Reply by dasbootbee** [reply]
I'm sorry -- I don't understand what you're asking me to test.

**Reply by krystian3w** [reply]
`##.customScrollBar > div > div.full:not([id])` in Picker: https://github.com/gorhill/uBlock/wiki/Element-picker

**Reply by dasbootbee** [reply]
##.evj9g.ab0a0 seems to work.  It's one of the suggested ones when I use the picker to select one of the Ad "emails".

**Reply by krystian3w** [reply]
Such a filter with random classes is of little value.

**Reply by dasbootbee** [reply]
I tried pasting your suggested filter in the box at the top but it didn't do anything.

**Comment by mapx-** [comment]
As this kind of ads / complex pages have a high breakage potential I'll convert the issue to a discussion where the .. discussion & filter ideas can continue.

**Comment by peace2000** [comment]
For this version:

![kuva](https://github.com/uBlockOrigin/uAssets/assets/17256841/974c6b5f-6cb4-4d5a-a0c0-130a12430500)

(This alternative place for the ad can be accessed by changing a setting on this link https://outlook.live.com/mail/0/options/general/advertisement (at the bottom)

I think this would be safe:

`outlook.live.com##.Ogt7j`

Not sure if they switch element names often. If so, I'd go with this:

`outlook.live.com##div[class][tabindex][role="complementary"][data-app-section][aria-label][data-min-width][data-min-height][data-max-height] > div[id][class]:first-of-type`

**Reply by Yuki2718** [reply]
We already has `outlook.live.com##[role="listbox"] .Ogt7j` and on my end `##div[data-app-section="MessageList"] div[role="listbox"] .customScrollBar > div[style] > div[style^="position: absolute"]:first-child:has(i[data-icon-name="Delete"] + div + button)` is also hiding this.

**Reply by peace2000** [reply]
Those rules don't target that upper ad. Did you try to change you settings on this setting page: https://outlook.live.com/mail/0/options/general/advertisement

**Reply by Yuki2718** [reply]
No, you can remove `[role="listbox"] ` from the rule then.

**Reply by Procyon-b** [reply]
As the one who proposed this filter, and has a userstyle to regularly update for hotmail, I can tell you that the classnames are quite stable. Over 4 months before they change it is usual. I have an element whose classname lasted 2 years between 2019 and 2021.

**Reply by peace2000** [reply]
Changed to `outlook.live.com##.Ogt7j`.

**Reply by peace2000** [reply]
I don't understand what's the idea of this filter:

`outlook.live.com##.customScrollBar > div[class] > div[class]:empty`

**Reply by Procyon-b** [reply]
> I don't understand what's the idea of this filter:
> 
> `outlook.live.com##.customScrollBar > div[class] > div[class]:empty`

On initial loading of the page with the ad above the maillist, I've seen that the div structure evolves. Sometimes the div doesn't fully mutate to the `.Ogt7j` form.
I don't remember exactly how it looked, but this filter seems similar to what I was seeing/trying.

**Reply by peace2000** [reply]
I cleaned all the old entries as they seemed obsolete.
